src/diffusers/schedulers/scheduling_lms_discrete.py

```diff
@@ -215,7 +210,18 @@ def step(
 
         if isinstance(timestep, torch.Tensor):
             timestep = timestep.to(self.timesteps.device)
-        step_index = (self.timesteps == timestep).nonzero().item()
+        if (
```

Comment:
This could lead to some nasty silent bugs though if the "new" passed timesteps are ints instead of float no? Not sure whether this is a good idea tbh